frappe/public/js/frappe/router.js

```diff
@@ -469,19 +469,34 @@ frappe.router = {
 				return encodeURIComponent(String(a));
 			}
 		}).join("/");
-		let private_home = frappe.workspaces[`home-${frappe.user.name.toLowerCase()}`];
 
-		let workspace_name = private_home || frappe.workspaces["home"] ? "home" : "";
-		let is_private = !!private_home;
-		let first_workspace = Object.keys(frappe.workspaces)[0];
+		if (path_string) {
+			return "/app/" + path_string;
+		}
+
+		// Workspace
+		let private_home = `home-${frappe.user.name.toLowerCase()}`;
+		let default_page = null;
+		if (frappe.boot.user.default_workspace) {
+			default_page = frappe.router.slug(frappe.boot.user.default_workspace.name);
+		} else if (frappe.workspaces[private_home]) {
+			default_page = private_home;
```

Comment:
Something like this?

```suggestion
			default_page = "home";
```

(redirect to `/app/home` even if there is a `home-myUser` workspace)

Reply:
no it's `/app/private/home`

Reply:
this won't still handle other private workspaces :skull:

Reply:
https://github.com/frappe/frappe/blob/df9494322ae296ca6bc00c67f553f0f5c4554837/frappe/public/js/frappe/views/workspace/workspace.js#L354-L356

this won't work for private workspaces because of naming :hankey:

Reply:
Well I understand the issue better now, now that you've opened your fix PR.

You see, I mainly work on a _fork_ of Frappe where we already decoupled the `name` from the `title` of Workspaces… When I ported my change to Frappe I forgot to use `title` where I should have. Hopefully I didn't break too much stuff, right?